packages/react-strict-dom/src/native/modules/createStrictDOMComponent.js

```diff
@@ -142,12 +142,18 @@ export function createStrictDOMComponent<T, P: StrictProps>(
         nativeProps.experimental_layoutConformance = 'strict';
       }
 
-      // $FlowFixMe
-      let element = <NativeComponent {...nativeProps} />;
+      let element: React.Node =
+        typeof props.children === 'function' ? (
+          props.children(nativeProps)
+        ) : (
+          // $FlowFixMe
+          <NativeComponent {...nativeProps} />
+        );
 
       if (
-        nativeProps.children != null &&
-        typeof nativeProps.children !== 'string'
+        (nativeProps.children != null &&
+          typeof nativeProps.children !== 'string') ||
+        typeof props.children === 'function'
```

Comment:
redundant?

Reply:
useNativeProps doesn't return `children` (it made the types simpler), so the check if done against the original props

Reply:
I missed the `nativeProps` vs `props`. Sorry.